File: open_notebook/domain/activity.py
"""Activity domain model for customer touchpoint logging."""
from typing import Optional

from open_notebook.domain.base import ObjectModel


class Activity(ObjectModel):
    table_name = "activity"

    customer_id: str = ""
    activity_type: str = ""
    description: str = ""
    metadata: Optional[dict] = None
    actor: str = "system"

File: frontend/src/lib/types/activity.ts
export interface Activity {
  id: string
  customer_id: string
  activity_type: string
  description: string
  metadata: Record<string, unknown>
  actor: string
  created: string
  updated: string
}

export interface ActivityCreate {
  customer_id: string
  activity_type: string
  description: string
  metadata?: Record<string, unknown>
  actor?: string
}
